draft-ietf-mls-extensions.md

```diff
@@ -483,6 +483,106 @@ The `media_type` MAY be zero length, in which case, the media type of the
 `application_content` is interpreted as the first MediaType specified in
 `required_media_types`.
 
+## Message Streams
+
+### Description
+
+By default, for each leaf node, [MLS defines](https://www.rfc-editor.org/rfc/rfc9420.html#section-9-5.1.1) two symmetric hash ratchets. One is used for handshake messages, and the other is used for application messages.
+
+Some applications may require more than one application message ratchet. For example, an application may want to send messages with different delivery semantics or different policies. This section describes an extension to MLS which allows MLS clients to send multiple types of messages. Each message type has it's own independent key ratchet.
```

Comment:
I agree with TWal. A slightly less vague example would help a lot in the intro.

"For example, one message stream might be used for frequent bulk-priority messages, while another is used for infrequent urgent notifications. This would allow a client after a network outage to decrypt the urgent stream immediately while still decrypting bulk-priority messages."

What happens with the handshake messages is still not clear though. If there are lots of adds and joins in the group, the client will still need to process all of the commits and all the ratchets will need to change anyway right?